### PR TITLE
[greenhouse][opentelemetry] add opentelemetry pluginpreset

### PR DESCRIPTION
**Pull request by timojohlo**

### CODE REVIEW

**Comment by timojohlo** [comment]
Not sure if we need additional labels e.g. `greenhouse.sap/cluster-presets-enabled: "true"` and if `cluster-type:` is necessary?

File: system/greenhouse-ccloud/templates/opentelemetry-pluginpreset.yaml [hunk 1]
```diff
@@ -0,0 +1,65 @@
+{{- if .Values.openTelemetry.enabled -}}
+{{- range $r := $.Values.openTelemetry.clusters }}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: otel-basic-auth-{{ $r.region }}
+  namespace: {{ .Release.Namespace }}
+data:
+  password: {{ required ".Values.openTelemetry.openSearchLogs.password missing" $r.openSearchLogs.password | b64enc }}
```
Comment (IvoGoman) [inline system/greenhouse-ccloud/templates/opentelemetry-pluginpreset.yaml:9]: The required text is misleading maybe ".Values.openTelemetry.clusters[].openSearchLogs.password missing"

File: system/greenhouse-ccloud/templates/opentelemetry-pluginpreset.yaml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  username: {{ required ".Values.openTelemetry.openSearchLogs.username missing" $r.openSearchLogs.username | b64enc }}
```
Comment (IvoGoman) [inline system/greenhouse-ccloud/templates/opentelemetry-pluginpreset.yaml:10]: The required text is misleading maybe ".Values.openTelemetry.clusters[].openSearchLogs.username missing"

File: system/greenhouse-ccloud/templates/opentelemetry-pluginpreset.yaml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  clusterOptionOverrides:
+  {{- range $r := $.Values.openTelemetry.clusters }}
+    clusterName: {{ $r.cluster }}
+    overrides:
+    - name: openTelemetry.OpenSearchLogs.endpoint
+      value: {{ $r.openSearchLogs.endpoint }}
+    - name: openTelemetry.OpenSearchLogs.username
+      value: {{ $r.openSearchLogs.username }}
+    - name: openTelemetry.OpenSearchLogs.password
+      value: {{ $r.openSearchLogs.password }}
```
Comment (IvoGoman) [inline system/greenhouse-ccloud/templates/opentelemetry-pluginpreset.yaml:34]: This should be `valueFrom` syntax to refer to the secret created above.

**Review by IvoGoman — CHANGES_REQUESTED** [review]

**Review by IvoGoman — APPROVED** [review]

**Pull request merged**